common/src/main/java/io/apicurio/registry/auth/KeycloakAuth.java

```diff
@@ -43,6 +43,7 @@ public KeycloakAuth(String serverUrl, String realm, String clientId, String clie
      */
     @Override
     public void apply(Map<String, String> requestHeaders) {
+        //FIXME this does not use the refresh token , it requests a new token for every request
         requestHeaders.put("Authorization", BEARER + this.keycloak.obtainAccessToken().getToken());
```

Comment:
Really? That seems bad. Should we open a GH issue for this?

Reply:
Yes the AuthzClient library which we're using in rest-client library has a very high-level API, which you cannot use for getting the access token by providing a refresh token. Probably worth a GH issue, yes.

Reply:
That said, in the client_credentials flow, and this is where `KeycloakAuth` is involved, the [specification](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.3) says to not include refresh tokens. Since there's no user interaction, and the client can receive an access token with its own credentials, there's no need to store an access token.

The only improvement for me here is perhaps to store the access token and re-use it or request a new one if it's expired, but, as I said, and as is stated in the spec, the refresh token part makes little sense in the `client_credentials` flow.

Reply:
It does.  The improvement then is clearly to cache the access token as long as possible.  Only get a new token when needed (due to expiration).  That seems like a pretty important improvement.

Reply:
Yep, [this](https://github.com/Apicurio/apicurio-registry/pull/1685) should address that issue.

Reply:
Can we resolve this thread then?  :)

Reply:
Unless @famartinrh has any objections yes, I think we can resolve this conversations